sympy/geometry/ellipse.py

```diff
@@ -695,27 +692,64 @@ def is_tangent(self, o):
         True
 
         """
-        inter = None
+        if isinstance(o, Point2D):
+            return o in self
         if isinstance(o, Ellipse):
-            inter = self.intersection(o)
-            if isinstance(inter, Ellipse):
+            if o in self:
+                return False
+            elif len(self.intersection(o)) == 1:
+                return True
+            elif len(self.intersection(o)) == 2 and \
+                    ((self.bounds[0] == o.bounds[0] and self.bounds[2] == self.bounds[2]) or
+                         (self.bounds[1] == o.bounds[1] and self.bounds[3] == self.bounds[3])):
+                return True
+            else:
+                return False
+        elif isinstance(o, Line2D):
+            if len(self.intersection(o)) == 1:
+                return True
+            else:
                 return False
-            return (inter is not None and len(inter) == 1
-                    and isinstance(inter[0], Point))
-        elif isinstance(o, LinearEntity):
-            inter = self.intersection(o)
-            if inter is not None and len(inter) == 1:
-                return inter[0] in o
+        elif isinstance(o, Ray2D):
+            if len(self.intersection(o)) == 1 and not any([self.encloses_point(point) for point in o.points]):
+                return True
             else:
                 return False
-        elif isinstance(o, Polygon):
-            c = 0
-            for seg in o.sides:
-                inter = self.intersection(seg)
-                c += len([True for point in inter if point in seg])
-            return c == 1
+        elif isinstance(o, (Segment2D, Polygon)):
+            segments = o.sides if isinstance(o, Polygon) else [o]
+            any_intersect = False
+            for side in segments:
```

Comment:
without reading too closely, I am guessing this would indicate that a circumcircle is tangent to its enclosed trianlge. If so, that' not right. The slope is not defined at the vertices.